browser/src/libs/github/util.tsx

```diff
@@ -182,29 +182,30 @@ function getDiffResolvedRevFromPageSource(pageSource: string, isPullRequest: boo
 }
 
 /**
- * Scrapes the branch name from the branch select menu element
- * present on GitHub blob pages.
+ * Returns the file path for the current page. Must be on a blob page.
+ *
+ * Implementation details:
+ *
+ * - This scrapes the file path from the "Copy path" button on GitHub blob
+ *   pages:
+ *
+ *     <clipboard-copy value="foo/bar.go"> Copy path </clipboard-copy>
+ *
+ * - We can't get the file path from the URL because the branch name can contain
+ *   slashes which make the boundary between the branch name and file path
+ *   ambiguous. For example:
+ *
+ *     https://github.com/sourcegraph/sourcegraph/blob/bext/release/cmd/frontend/internal/session/session.go
+ *
+ * TODO ideally, this should only scrape the code view itself.
  */
-export function getBranchName(): string {
-    const branchSelectMenu = document.querySelector('.branch-select-menu') as HTMLElement
-    if (!branchSelectMenu) {
-        throw new Error('Could not find .branch-select-menu')
-    }
-    const cssTruncateTarget = branchSelectMenu.querySelector('span.css-truncate-target') as HTMLElement
-    if (!cssTruncateTarget) {
-        throw new Error('Could not find span.css-truncate-target')
-    }
-    if (!cssTruncateTarget.innerText.endsWith('…')) {
-        // The branch name is not truncated
-        return cssTruncateTarget.innerText
-    }
-    // When the branch name is truncated, it is stored in full
-    // in the select menu button's title attribute.
-    const selectButton = cssTruncateTarget.closest('.select-menu-button') as HTMLElement
-    if (!selectButton) {
-        throw new Error('Could not find .select-menu-button')
-    }
-    return selectButton.title
+export function getFilePath(): string | undefined {
+    return document.evaluate(
```

Comment:
I see why XPath is used here but it should be documented why - in general I'd like to avoid it because we can't assume people on the team understand it. Asserting the content would also break if GitHub had different translations (BitBucket does for example, so this shouldn't become standard practice).

Could you benchmark that the performance of this is not worse than the alternative `document.querySelectorAll().find()`?

```suggestion
    // Using XPath here instead of CSS selectors because the only way to know is the content of the button
    // GitHub is always in English
    return document.evaluate(
```

Reply:
Switched to CSS `clipboard-copy[value]` to avoid relying on the text being in English and to avoid XPath.